### PR TITLE
Cf 866 more port drayage

### PR DESCRIPTION
**Pull request by john-chrosniak**
# PR Details
## Description

This PR adds additional functionality needed for the C1T port drayage demo. This includes:

1. Making the vehicle's `cmv_id` a configurable parameter
2. Adding a check to ensure the vehicle only acts on a received `MobilityOperation` message if the requested `cmv_id` matches the vehicle's
3. Incorporates logic to track whether or not the vehicle is carrying cargo and, if so, also track the `cargo_id`

Related PR: [c1t_bringup #20](https://github.com/usdot-fhwa-stol/c1t_bringup/pull/20)

## Related Jira Key

[CF-866](https://usdot-carma.atlassian.net/browse/CF-866)

## Motivation and Context

The carma_nav2_port_drayage_demo package was recently updated in [CF-852](https://usdot-carma.atlassian.net/browse/CF-852)  to handle processing of received port drayage-related Mobility Operation messages from infrastructure, and to publish port drayage Mobility Operation messages upon arrival at a destination. This implementation has some hardcoded values, and the package should be updated to better handle some of these variables, while also making some configurable if necessary.

## How Has This Been Tested?

This was tested using the turtlebot simulator and sending Mobility Operation messages to instruct the vehicle to pickup and dropoff cargo. Upon picking up cargo, the vehicle acknowledged that it was carrying cargo with the correct cargo id. After dropping off the cargo, it then acknowledged that is was no longer carrying cargo. Additionally, if Mobility Operation messages were sent with the wrong `cmv_id`, the vehicle disregarded the message.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[CF-866]: https://usdot-carma.atlassian.net/browse/CF-866?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CF-852]: https://usdot-carma.atlassian.net/browse/CF-852?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/turtlebot_port_drayage_test.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
```
Comment (willjohnsonk) [inline test/turtlebot_port_drayage_test.sh:2]: I would add a brief description of what this script is doing as a comment

File: carma_nav2_port_drayage_demo/src/port_drayage_demo.cpp [hunk 2]
```diff
@@ -56,16 +101,12 @@ auto PortDrayageDemo::on_configure(const rclcpp_lifecycle::State & /* state */)
 auto PortDrayageDemo::on_activate(const rclcpp_lifecycle::State & /* state */)
   -> nav2_util::CallbackReturn
 {
-  RCLCPP_INFO(get_logger(), "Activating");
```
Comment (willjohnsonk) [inline carma_nav2_port_drayage_demo/src/port_drayage_demo.cpp:59]: Is there any reason that the logger was removed at intermediate steps? Seems to be gone in multiple places

**Review by willjohnsonk — COMMENTED** [review]
I see in the original port drayage plugin in CARMA Platform that there's functionality that updates the Port Drayage web ui with popup text, does this need to be added in?
https://github.com/usdot-fhwa-stol/carma-platform/blob/dcfa36556d79471ada6c961cb5688c66a392603c/port_drayage_plugin/src/port_drayage_worker.cpp#L119

File: carma_nav2_port_drayage_demo/src/port_drayage_demo.cpp (hunk 2 above)
Comment (john-chrosniak) [reply carma_nav2_port_drayage_demo/src/port_drayage_demo.cpp:59]: The Nav2 lifecycle manager has logging built in, so this was basically a duplicate log

**Review by afrye51 — APPROVED** [review]
Looks good to me

**Pull request merged**